internal/cmd/auth.go

```diff
@@ -212,15 +223,25 @@ func authenticationHTTPMount(
 	}
 
 	if cfg.Methods.OIDC.Enabled {
-		oidcmiddleware := authoidc.NewHTTPMiddleware(cfg.Session)
+		oidcmiddleware := method.NewHTTPMiddleware(cfg.Session)
 		muxOpts = append(muxOpts,
-			runtime.WithMetadata(authoidc.ForwardCookies),
+			runtime.WithMetadata(method.ForwardCookies),
 			runtime.WithForwardResponseOption(oidcmiddleware.ForwardResponseOption),
 			registerFunc(ctx, conn, rpcauth.RegisterAuthenticationMethodOIDCServiceHandler))
 
 		middleware = append(middleware, oidcmiddleware.Handler)
 	}
 
+	if cfg.Methods.Github.Enabled {
+		githubMiddleware := method.NewHTTPMiddleware(cfg.Session)
+		muxOpts = append(muxOpts,
+			runtime.WithMetadata(method.ForwardCookies),
```

Comment:
I would if we should add a method to `AuthenticationConfig` for whether or not session support needs enabling.

So we can do something like:
```go
if cfg.SessionEnabled() {
    muxOpts = append(muxOpts, runtime.WithMetadata(method.ForwardCookies))
}
```

It could perform the same operation as what happens in the validate method right now:
https://github.com/flipt-io/flipt/blob/2569b5511b0ae26e234038b7d8a05f9cd50797b9/internal/config/authentication.go#L117-L120

Reply:
The idea being that we dont append the middleware twice when both are enabled.